frontend/apps/crates/components/src/profile_image/actions.rs

```diff
@@ -0,0 +1,50 @@
+use std::rc::Rc;
+
+use crate::image::upload::upload_image;
+use dominator::clone;
+use shared::{
+    api::endpoints,
+    domain::image::{
+        user::{UserImageCreatePath, UserImageCreateRequest},
+        ImageId, ImageSize,
+    },
+    media::MediaLibrary,
+};
+use utils::{prelude::ApiEndpointExt, unwrap::UnwrapJiExt};
+use web_sys::File;
+
+use super::{ProfileImage, ImageIdOrFile};
+
+impl ProfileImage {
+    pub fn apply_changes(self: &Rc<Self>) {
+        let state = self;
+        state.loader.load(clone!(state => async move {
+            let image_id = match state.image.get_cloned() {
+                None => None,
+                Some(ImageIdOrFile::ImageId(image_id)) => Some(image_id),
+                Some(ImageIdOrFile::File(image_file)) => {
+                    Some(upload_profile_image(image_file).await.unwrap_ji())
+                },
+            };
+
+            (state.config.save_changes)(image_id);
```

Comment:
Think the popup should be closed after saving, something like `state.popup_open.set(false)`